src/board_builder/board_builder.py

```diff
@@ -0,0 +1,258 @@
+                    self._expand_matrix()
+
+        for detector_name in detector_data:
+            for markerSnapshot in detector_data[detector_name]:
+                corners_list: list[list[int]] = []
+                for corner in markerSnapshot.corner_image_points:
+                    corners_list.append([corner.x_px, corner.y_px])
+                marker_corners = MarkerCorners(
+                    detector_label=detector_name,
+                    marker_id=int(markerSnapshot.label),
+                    points=corners_list,
+                    timestamp=timestamp
+                )
+                self.pose_solver.add_marker_corners([marker_corners])
+
+        target_poses = self.pose_solver.get_target_poses()
+        self.target_poses = target_poses
+        visible_markers = []
+        for pose in target_poses:
+            visible_markers.append(pose.target_id)
+            if pose.target_id not in list(self._index_to_marker_uuid.values()):
+                self._index_to_marker_uuid[self._matrix_index] = pose.target_id
+                self._matrix_index += 1
+        self._visible_markers = visible_markers
+
+
+    ### PUBLIC METHOD ###
+    def locate_reference_markers(self, detector_data):
+        detector_data = self._filter_data(detector_data)
+        if not detector_data:
+            return
+        self.detector_poses = []
+        timestamp = datetime.datetime.now()
+        for detector_name in detector_data:  # O(N^3) is pretty bad time complexity
+            for markerSnapshot in detector_data[detector_name]:
+                corners_list: list[list[int]] = []
+                for corner in markerSnapshot.corner_image_points:
+                    corners_list.append([corner.x_px, corner.y_px])
+                marker_corners = MarkerCorners(
+                    detector_label=detector_name,
+                    marker_id=int(markerSnapshot.label),
+                    points=corners_list,
+                    timestamp=timestamp
+                )
+                self.pose_solver.add_marker_corners([marker_corners])
+
+        new_detector_poses = self.pose_solver.get_detector_poses()
+        for pose in new_detector_poses:
+            if pose.target_id not in self._detector_poses_average:
+                self._detector_poses_average[pose.target_id] = PoseLocation(pose.target_id)
+            self._detector_poses_average[pose.target_id].add_matrix(np.array(pose.object_to_reference_matrix.values).reshape(4, 4), str(timestamp))
+        for label in self._detector_poses_average:
+            pose = Pose(
+                target_id=label,
+                object_to_reference_matrix=self._detector_poses_average[label].get_pose().object_to_reference_matrix,
+                solver_timestamp_utc_iso8601=str(timestamp)
+            )
+            self.detector_poses.append(pose)
+        self.pose_solver.set_detector_poses(self.detector_poses)
```

Comment:
1. Yes, this is mainly to eliminate outliers (therefore it would make sense to use median). 
2. It is taking in data as long as the "Locate Reference" toggle button is on.
3. Since the goal of this part is to eliminate outliers, it would make sense to use median.